File: DayOne/Java/LevelTwo/LevelTwoTest.java
import org .junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

public class LevelTwoTest{

    @Test
    public void testThatIhaveArrayAndItsReturnsDuplicateElements(){

    int [] array = {1, 2, 3, 2, 4, 3};

    int [] arrayCheck = {2, 3};

    int [] actual = LevelTwo.duplicateCheck(array);

    assertArrayEquals(actual,arrayCheck);
    }

    @Test
    public void testThatIhaveArrayAndItReturnsTheDuplicateElementAndIndex(){

    int [] array = {-11, -9, 3, -9, 2, -11};

    Object [][] arrayCheck = {{-11, new int []{0, 5} }, {-9, new int []{1, 3} }};

    Object [][] actual = LevelTwo.getArrayIndexAndDuplicate(array);

    assertArrayEquals(actual,arrayCheck);
    }

    @Test
    public void testThatIhaveArrayAndIMoveAllZeroElementToTheEndOfTheArray(){
    
    int [] array = {5,0,3,0,2,0};

    int [] arrayCheck = {5,3,2,0,0,0};

    int [] actual = LevelTwo.moveAllZerosIn(array);

    assertArrayEquals(actual,arrayCheck);
    }
}

File: DayOne/Python/LevelTwo/level_two_test.py
from unittest import TestCase

from level_two import *

class LevelTwoTest(TestCase):

    def test_that_i_have_an_array_and_i_return_an_array_of_its_duplicate(self):

        array = [1, 2, 3, 2, 4, 3]

        array_check = [2, 3]

        self.assertEqual(duplicate_check(array),array_check)

    def test_that_i_have_an_array_and_i_return_array_with_its_duplicates_and_indexes(self):

        array = [-11, -9, 3, -9, 2, -11]

        array_check = [ [-11, [0, 5] ], [-9, [1, 3] ] ]

        self.assertEqual(get_array_index_and_duplicate(array),array_check)

    def test_that_i_have_an_array_and_i_move_zero_elements_to_the_end_of_the_array(self):

        array = [5, 0 , 3, 0, 2, 0]

        array_check = [5, 3, 2,0,0,0]

        self.assertEqual(move_all_zeros_in(array),array_check)
